components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/EndpointResource.java

```diff
@@ -196,12 +196,12 @@ private void changeEndpointStatus(org.apache.axis2.context.MessageContext axis2M
                 jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, endpointName + " is switched On");
             } else {
                 LOG.warn("Specified Status: " + status + " is not valid");
-                jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, "Provided state is not valid");
+                jsonResponse = Utils.createJsonError("Provided state is not valid", axis2MessageContext, "400");
             }
             Utils.setJsonPayLoad(axis2MessageContext, jsonResponse);
         } else {
-            LOG.warn("Specified Endpoint: " + endpointName + " does not exist");
-            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Endpoint does not exist"));
+            Utils.setJsonPayLoad(axis2MessageContext,  Utils.createJsonError("Endpoint does not exist",
+                    axis2MessageContext, "403"));
```

Comment:
Wouldn't it be better if we set 404 instead of 403 here? WDYT?

Reply:
404 is used when Request-URI is unavailable.
Ideally web resource is available right? it's only the endpoint the user is trying to deactivate is missing. WDYT?

Reply:
Ideally the request accessed here (the endpoint) should have been in the request path and the request will match the spec of 404 in that case. so I think 404 is more applicable. 403, is also fine since we are giving the reason for forbidding the request.

Reply:
Changed the non-existing resources to 404 SC